src/shogun/multiclass/ShareBoost.cpp

```diff
@@ -53,8 +54,9 @@ bool CShareBoost::train_machine(CFeatures* data)
 
 	if (m_nonzero_feas <= 0)
 		SG_ERROR("Set a valid (> 0) number of non-zero features to seek before training\n")
-	if (m_nonzero_feas >= dynamic_cast<CDenseFeatures<float64_t>*>(m_features)->get_num_features())
-		SG_ERROR("It doesn't make sense to use ShareBoost with num non-zero features >= num features in the data\n")
+	if (m_nonzero_feas > num_features)
+		SG_ERROR("Number of non-zero features (%d) cannot be larger than number of features (%d) in the data\n",
+				m_nonzero_feas, num_features)
 
 	m_fea = dynamic_cast<CDenseFeatures<float64_t> *>(m_features)->get_feature_matrix();
```

Comment:
there are still two dynamic casts in here.
`dynamic_cast<CDenseFeatures<float64_t> *>(m_features)` This is what I meant, not the num_features (though that can stay also)